iceoryx_posh/test/moduletests/test_popo_client_port.cpp

```diff
@@ -0,0 +1,552 @@
+
+    EXPECT_FALSE(sut.getResponse().has_error());
+
+    EXPECT_FALSE(sut.hasNewResponses());
+}
+
+TEST_F(ClientPort_test, HasLostResponsesSinceLastCallWithoutLosingResponsesReturnsFalse)
+{
+    auto& sut = clientPortUserWithConnectOnCreate;
+    EXPECT_FALSE(sut.hasLostResponsesSinceLastCall());
+}
+
+TEST_F(ClientPort_test, HasLostResponsesSinceLastCallWithoutLosingResponsesAndQueueFullReturnsFalse)
+{
+    auto& sut = clientPortUserWithConnectOnCreate;
+
+    for (auto i = 0U; i < QUEUE_CAPACITY; ++i)
+    {
+        constexpr uint32_t USER_PAYLOAD_SIZE{10};
+        auto sharedChunk = getChunkFromMemoryManager(USER_PAYLOAD_SIZE, sizeof(ResponseHeader));
+        if (!chunkQueuePusherWithConnectOnCreate.push(sharedChunk))
+        {
+            chunkQueuePusherWithConnectOnCreate.lostAChunk();
+        }
+    }
+
+    EXPECT_FALSE(sut.hasLostResponsesSinceLastCall());
+}
+
+TEST_F(ClientPort_test, HasLostResponsesSinceLastCallWithLosingResponsesReturnsTrue)
+{
+    auto& sut = clientPortUserWithConnectOnCreate;
+
+    for (auto i = 0U; i < QUEUE_CAPACITY + 1; ++i)
+    {
+        constexpr uint32_t USER_PAYLOAD_SIZE{10};
+        auto sharedChunk = getChunkFromMemoryManager(USER_PAYLOAD_SIZE, sizeof(ResponseHeader));
+        if (!chunkQueuePusherWithConnectOnCreate.push(sharedChunk))
+        {
+            chunkQueuePusherWithConnectOnCreate.lostAChunk();
+        }
+    }
+
+    EXPECT_TRUE(sut.hasLostResponsesSinceLastCall());
+}
+
+TEST_F(ClientPort_test, HasLostResponsesSinceLastCallReturnsFalseAfterPreviouslyReturningTrue)
+{
+    auto& sut = clientPortUserWithConnectOnCreate;
+
+    for (auto i = 0U; i < QUEUE_CAPACITY + 1; ++i)
+    {
+        constexpr uint32_t USER_PAYLOAD_SIZE{10};
+        auto sharedChunk = getChunkFromMemoryManager(USER_PAYLOAD_SIZE, sizeof(ResponseHeader));
+        if (!chunkQueuePusherWithConnectOnCreate.push(sharedChunk))
+        {
+            chunkQueuePusherWithConnectOnCreate.lostAChunk();
+        }
+    }
+
+    EXPECT_TRUE(sut.hasLostResponsesSinceLastCall());
+    EXPECT_FALSE(sut.hasLostResponsesSinceLastCall());
+}
+
+TEST_F(ClientPort_test, ConditionVariableInitiallyNotSet)
+{
+    auto& sut = clientPortUserWithConnectOnCreate;
+    EXPECT_FALSE(sut.isConditionVariableSet());
+}
+
+TEST_F(ClientPort_test, SettingConditionVariableWithoutConditionVariablePresentWorks)
+{
+    iox::popo::ConditionVariableData condVar{"hypnotoad"};
+    constexpr uint32_t NOTIFICATION_INDEX{1};
+
+    auto& sut = clientPortUserWithConnectOnCreate;
+    sut.setConditionVariable(condVar, NOTIFICATION_INDEX);
+
+    EXPECT_TRUE(sut.isConditionVariableSet());
+}
+
+TEST_F(ClientPort_test, UnsettingConditionVariableWithConditionVariablePresentWorks)
+{
+    iox::popo::ConditionVariableData condVar{"brain slug"};
+    constexpr uint32_t NOTIFICATION_INDEX{2};
+
+    auto& sut = clientPortUserWithConnectOnCreate;
+    sut.setConditionVariable(condVar, NOTIFICATION_INDEX);
+
+    sut.unsetConditionVariable();
```

Comment:
Can setting and unsetting happen concurrently? I think this should be described in doxygen.

Reply:
No, it can't. None of the ports can be used concurrently. I don't know it this is stated somewhere but can add it to the brief description of the client port

Reply:
Added a note to the class description.